File: CMP326Fall2019MidtermProject1/gameinterfaces/Player.java
package gameinterfaces;


/**
 * 
 * A class that implements this interface should have the following attributes:
 * <ul>
 * <li>healAmount: int</li>
 * </ul>
 * 
 * The implementing class should only have the following constructor:
 * 
 * public Player(String name, String description, int hitPoints, int damage, int healAmount)
 * 
 */

public interface Player extends Creature {

	/**
	 * 
	 * This method will increase the <code>hitPoints</code> for <code>this Player</code> by <code>this.healAmount</code>.
	 * <br>
	 * <br>
	 * This method will also display the message:
	 * <br>
	 * <br>
	 * "<code>this.name</code> is healed by <code>this.healAmount</code>"
	 * 
	 */
	public void heal();
	
}

File: Qt_workspace/rpg_txt_game/player.cpp
#include "player.h"

Player::Player(){
    Creature();
}
Player::Player(string name, string description, int hitPoints,
               int damage, int healAmount)
{
    Creature(name, description, hitPoints, damage);
    this->healAmount = healAmount;
}

void Player::heal(){
    this->setHitPoints(this->healAmount);
    cout << this->getName() << "is healed by " << this->healAmount << endl;
}

